src/components/map/wr-map-legend.tsx

```diff
@@ -1,28 +1,27 @@
+import { MapPin } from "lucide-react";
+
+import type { WrLegendItem } from "@/lib/types";
 import { cn } from "@/lib/utils";
 
 interface LegendProps {
-  items: {
-    text: string;
-    color: string;
-  }[];
+  items: WrLegendItem[];
   className?: string;
 }
 
-export function Legend({ items, className }: LegendProps) {
+export function WrMapLegend({ items, className }: LegendProps) {
   return (
     <div
       className={cn(
-        "mx-auto grid w-[90%] grid-cols-2 gap-x-28 gap-y-2 sm:w-fit sm:gap-x-4",
+        "mx-auto grid w-[90%] grid-cols-2 gap-x-5 gap-y-2 sm:grid-cols-[repeat(3,minmax(0,20vw))] md:w-max",
```

Comment:
Może lepiej po prostu tak
```suggestion
        "grid w-full grid-cols-1 gap-x-5 gap-y-2 sm:grid-cols-3",
```